client/src/Components/Pages/Prescriptions/Prescriptions.tsx

```diff
@@ -0,0 +1,86 @@
+
+import { useEffect, useState } from "react"
+import { useQuery } from "react-query"
+import { useHistory } from "react-router-dom"
+import BackArrow from "../../../assets/utils/back-arrow.svg"
+import { useAuth } from "../../../Contexts/Auth.context"
+import { getTranslatedText } from "../../../services/api.translate"
+import client from "../../../services/graphqlService"
+import queries from "../../../services/graphqlService/queries"
+import { ConsultationInfo } from "../../../types"
+import Spinner from "../../Globals/Spinner/Spinner"
+import Prescription from "./Prescription"
+
+interface PrescriptionData {
+  date: string;
+  medicine: string;
+  dose: string;
+  frequency: string;
+  doctor: string;
+}
+
+const parseDate = (date: string) =>
+  new Date(Date.parse(date)).toLocaleString().split(',')[0];
+
+const Prescriptions = () => {
+  const history = useHistory()
+  const {user} = useAuth();
+  const [userId, userLang] = [user!.id, user!.language]
+  const [prescriptions, setPrescriptions] = useState<PrescriptionData[]>([])
+
+  const translateFreq = async (freq: string) => {
+    return await getTranslatedText(freq, "en", userLang);
+  }
+
+  const {data, isLoading} = useQuery([userId, "consultations"],
+    async () => await client.request(queries.getPatientConsultations, {id: userId}));
+
+  useEffect(() => {
+    if (data) {
+      (async () => {
+        const parsedPrescr = getPrescriptions(data.getPatientConsultations)
+        parsedPrescr.forEach(async  p => {p.frequency = await translateFreq(p.frequency)})
+        setPrescriptions(parsedPrescr)
+      })();
```

Comment:
Not a necessary change - I'd put this async callback in the useQuery's onSuccess option